components/board/mcp3008helper/mcp3008.go

```diff
@@ -37,29 +38,32 @@ func (config *MCP3008AnalogConfig) Validate(path string) error {
 	return nil
 }
 
-func (mar *MCP3008AnalogReader) Read(ctx context.Context, extra map[string]interface{}) (value int, err error) {
+func (mar *MCP3008AnalogReader) Read(ctx context.Context, extra map[string]interface{}) (
+	value int, analogRange board.AnalogRange, err error,
+) {
 	var tx [3]byte
 	tx[0] = 1                            // start bit
 	tx[1] = byte((8 + mar.Channel) << 4) // single-ended
 	tx[2] = 0                            // extra clocks to receive full 10 bits of data
 
 	bus, err := mar.Bus.OpenHandle()
 	if err != nil {
-		return 0, err
+		return 0, board.AnalogRange{}, err
 	}
 	defer func() {
 		err = multierr.Combine(err, bus.Close())
 	}()
 
 	rx, err := bus.Xfer(ctx, 1000000, mar.Chip, 0, tx[:])
 	if err != nil {
-		return 0, err
+		return 0, board.AnalogRange{}, err
 	}
 	// Reassemble the 10-bit value. Do not include bits before the final 10, because they contain
 	// garbage and might be non-zero.
 	val := 0x03FF & ((int(rx[1]) << 8) | int(rx[2]))
 
-	return val, nil
+	// returning blank analog range since mcp3008 will be removed soon.
+	return val, board.AnalogRange{}, nil
```

Comment:
I like the comment. However, if you want to return data in the meantime, the min is 0, the max is 1024, and the step size is 1.

Reply:
how are you getting those values? the range depends on vref

Reply:
I'm still mixing up voltages and the raw values. Put what you just wrote in the comment in the code, so people like me don't make this mistake again.